app/models/rule.rb

```diff
@@ -8,11 +8,13 @@ class Rule < ActiveRecord::Base
     points_for_win:       0,
     points_for_loss:      1,
     points_for_tie:       2,
-    points_for_time_win:  3
```

Comment:
Have you considered using a reduce function here?

Every time a new value is added the programmer has to add both a key (name) _and_ an index that is incremented by `n`. Adding the index is an extra unnecessary step.

The index is an implementation detail that has no bearing on the domain: I can't go to James and say "points for a win has a 0 index!" 'cos the index has nothing to do with Netrunner; I _can_ say "we've got points for a win, a tie, a loss and stuff like that" 'cos it's the names are the important thing and they do have meaning in the domain of Netrunner.

```ruby
KEYS = [
  :points_for_win,
  :points_for_loss,
  :points_for_tie,
  :points_for_time_win,
  :points_for_participation,
  :max_points_for_participation
].each_with_index.inject({}) { |memo, (key, index) | memo[key] = index; memo }
```

Using a reduction here means that when a new value is required the programmer just needs to add a new name to the source list: the index will be automatically derived.